### PR TITLE
StickySection component

### PR DESCRIPTION
**Pull request by ogupte**
This component is required for Sticky Headers to be implemented for Table.

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- ~~One core team UX approval~~

### CODE REVIEW

**Comment by danhahn** [comment]
I found that at narrow window sizes like 700px or less the sticky headers start to miss behave.  In some cases they did not stick at all in other cases they stuck but unstuck sooner than expected.  The more narrow the page the quicker it will get unstuck.

**Comment by ogupte** [comment]
correct, but that is expected since the `lowerBound` prop values are constant. There is a another component which will pass in a variable value for `lowerBound` and do what you expect.

File: src/components/StickySection/StickySection.jsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+		const {
+			isAboveFold,
+			containerRect,
+		} = this.state;
+
+		return (
+			<div
+				{...passthrus}
+				className={boundClassNames('&', '&-base', className)}
```
Comment (jondlm) [inline src/components/StickySection/StickySection.jsx]: is `-base` necessary?

Comment (ogupte) [inline src/components/StickySection/StickySection.jsx]: probably not

**Pull request merged**